### ISSUE TITLE
Cannot create AAAA alias for apigaway target

### ISSUE THREAD

**Issue by tomlaube** [aws/aws-cdk#3954]
### What is the problem?

AAAA record alias fails to validate it's input target correctly in python. Incorrect object arguent type is identified by jsii.
### Reproduction Steps

```python
from aws_cdk.aws_route53 import ARecord, AaaaRecord, HostedZone, AddressRecordTarget
zone = HostedZone.from_lookup(self, "Domain", domain_name=domain_name)
AaaaRecord(self, 'AaaaAlias', zone=zone,
           target=AddressRecordTarget.from_alias(ApiGatewayDomain(domain_name=the_lambda_api_gateway.domain_name)))

```
### Verbose Log

```
jsii.errors.JavaScriptError: 
  Error: Object of type @aws-cdk/aws-route53.RecordTarget is not convertible to @aws-cdk/aws-route53.AddressRecordTarget
      at Object.deserialize (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:12263:27)
      at Kernel._toSandbox (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:7031:61)
      at mapValues (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:12218:29)
      at mapValues (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:12447:27)
      at Object.deserialize (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:12214:20)
      at Kernel._toSandbox (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:7031:61)
      at xs.map (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:7084:33)
      at Array.map (<anonymous>)
      at Kernel._boxUnboxParameters (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:7084:19)
      at Kernel._wrapSandboxCode (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:7134:19)
      at Kernel._create (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6666:26)
      at Kernel.create (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6419:21)
      at KernelHost.processRequest (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6191:28)
      at KernelHost.run (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6137:14)
      at processRequest (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6137:45)
      at KernelHost.processRequest (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6233:16)
      at KernelHost.run (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6137:14)
      at processRequest (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6137:45)
      at KernelHost.processRequest (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6233:16)
      at KernelHost.run (/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:6137:14)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "cdk_app.py", line 35, in <module>
    api_stack = ApiStack(app, f'exampple-api-{stage}', env=env, **kwargs)
  File "/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_runtime.py", line 66, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/home/tom/workspace/work/exampple-api/cdk_infra/api_stack.py", line 92, in __init__
    target=AddressRecordTarget.from_alias(ApiGatewayDomain(domain_name=the_lambda_api_gateway.domain_name)))
  File "/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_runtime.py", line 66, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/aws_cdk/aws_route53/__init__.py", line 2602, in __init__
    jsii.create(AaaaRecord, self, [scope, id, props])
  File "/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_kernel/__init__.py", line 208, in create
    overrides=overrides,
  File "/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_kernel/providers/process.py", line 331, in create
    return self._process.send(request, CreateResponse)
  File "/home/tom/.cache/pypoetry/virtualenvs/exampple-api-v4PV-RTY-py3.7/lib/python3.7/site-packages/jsii/_kernel/providers/process.py", line 316, in send
    raise JSIIError(resp.error) from JavaScriptError(resp.stack)
jsii.errors.JSIIError: Object of type @aws-cdk/aws-route53.RecordTarget is not convertible to @aws-cdk/aws-route53.AddressRecordTarget

```
### Environment

  - **CDK CLI Version:**  1.6.1 (build a09203a)
  - **Module Version:**  1.6.1
  - **OS:**  Linux
  - **Language:**  Python

**Comment by TrueBrain** [comment]
I just hit the same issue (with CDK 1.19.0). Seems `AaaaRecord` is still using the deprecated `AddressRecordTarget`.

https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-route53/lib/record-set.ts#L174

What seems to work for now is to us the  `alias_target` parameter from `__init__` instead of the `from_alias()` function.
Because `AddressRecordTarget` is a child of `RecordTarget`, `from_alias()` seems like a valid choice, but it returns a `RecordTarget`, causing the above error.

So in other words:
`AddressRecordTarget(alias_target=ApiGatewayDomain(domain_name=the_lambda_api_gateway.domain_name))`
